### PR TITLE
Update email validation

### PR DESCRIPTION
**Pull request by prachigarg19**
Implement zod email validation in place of current validation to obtain a common approach throughout the webapp

### CODE REVIEW

**Comment by mariahosfeld** [comment]
Scope needs to be expanded before release.

File: src/utils/isEmailValid.ts
Comment (mohitb35) [inline src/utils/isEmailValid.ts:1]: 1. Has 'zod' been imported? I see a warning - Cannot find module 'zod' or its corresponding type declarations.
2. The INVALID_DOMAIN_LIST is not being considered any more.

**Review by mohitb35 — CHANGES_REQUESTED** [review]
@prachigarg19 I'm not sure if this PR was ready for review, but it currently does not work.

**Review by mohitb35 — APPROVED** [review]

File: src/utils/isEmailValid.ts [hunk 1]
```diff
@@ -1,22 +1,20 @@
+import { z } from 'zod';
```
Comment (norbertschuler) [inline src/utils/isEmailValid.ts:1]: `zod` is imported only as a dependency from the `next` package in our `package-lock.json`. But I would suggest adding all packages used in our code directly in our `package.json` with `npm add zod` and not rely on it being loaded by some dependency!

**Review by norbertschuler — CHANGES_REQUESTED** [review]

File: src/utils/isEmailValid.ts (hunk 1 above)
Comment (norbertschuler) [reply src/utils/isEmailValid.ts:1]: Btw. you did add the package `zod` in https://github.com/Plant-for-the-Planet-org/planet-donations/pull/440 ;-)

Comment (norbertschuler) [reply src/utils/isEmailValid.ts:1]: Did that myself now.

**Review by mohitb35 — APPROVED** [review]

**Pull request merged**